woocommerce/woocommerce-ios
# Store creation M2: UI/UX updates

Parts of https://github.com/woocommerce/woocommerce-ios/issues/8108, https://github.com/woocommerce/woocommerce-ios/issues/8045


### Description


This PR includes UI/UX updates for the store creation flow:

- Polishes for the WPCOM plan screen:
  - Update icons (export the currency image from the design HyVloP5FipZzyPVenH2euI-fi-2369%3A102723&t=xGNLQ5txsVMW4lHY-0)
  - Accommodate long price text by splitting it into two lines (I also had to add `.minimumScaleFactor(0.3).lineLimit(2)` to auto-shrink the font size since the font size in the design is too large for my local currency)
  - Update refund info to be footnote style
  - Update the feature image to fit in the same size 18x18 (the new currency image is a different size from other icons)
- Domain selector initial query should trigger an API request immediately instead of lagging from the debounce
  - Please let me know if there's a better way to do this than concatenating two publishers (`first` and `debounce`)
- The first in-progress screen is now shown in the navigation controller instead of a separate modal, it feels less jarring this way
  - Because the `InProgressViewController` is now in a navigation stack, hiding the navigation bar is necessary (I tried setting `navigationController.setNavigationBarHidden` in `StoreCreationCoordinator` but the animation looked odd)

### Testing instructions


- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the initial in-progress screen should be shown briefly, and then navigates to the store name screen in the same navigation stack
- Enter some store name and tap `Continue` --> the domain selector should show a spinner right away while loading domain suggestions
- Select a domain of your choice and tap `Continue`
- Tap `Continue to payment` --> the WPCOM plan screen should look like the design


### Screenshots


WPCOM plan screen:

before | after
-- | --
![Simulator Screen Shot - iPhone 14 - 2022-12-01 at 09 48 02](https://user-images.githubusercontent.com/1945542/204946463-b43b8d2f-3d7e-4d89-8cf9-88830c2f6a37.png) | ![Simulator Screen Shot - iPhone 14 - 2022-12-01 at 09 43 14](https://user-images.githubusercontent.com/1945542/204946453-c7bf28ad-64d8-4431-8d8c-9460cfc10ac9.png)



---
